### PR TITLE
Enhanced player list demo

### PR DESCRIPTION
**Pull request by vermentor**

### CODE REVIEW

File: client/src/GamemasterInGame/Lobby.tsx [hunk 1]
```diff
@@ -59,20 +59,24 @@ export function Lobby({ rolesList }: LobbyProps) {
...
-              <Text className="mr-1">Night Order</Text>
+          ) : (
+            <Tabs.Trigger className="max-w-[200px] flex-1" value="roles">
+              <Text className="mr-1">
+                Roles(
+                {
+                  Object.values(
+                    characterSelectState.selectedRoles.value,
+                  ).filter((selected) => selected).length
+                }
```
Comment (vermentor) [inline client/src/GamemasterInGame/Lobby.tsx:77]: Small error was counting deselected roles in tab header.

File: client/src/GamemasterInGame/NightOrder.tsx [hunk 2]
```diff
@@ -1,34 +1,31 @@
...
+import { Button, Flex } from "@radix-ui/themes";
+import { useDefiniteGame } from "../store/GameContext";
+
+// interface NightOrderProps {
+//   playersToRoles: Record<string, Role>;
+//   deadPlayers: Record<string, boolean>;
+// }
 
 export function NightOrder() {
   const { game } = useDefiniteGame();
```
Comment (vermentor) [inline client/src/GamemasterInGame/NightOrder.tsx:13]: I'd tried passing the above interface props in, but playerToRoles was always null for whatever reason

File: client/src/GamemasterInGame/PlayerList.tsx [hunk 3]
```diff
@@ -55,37 +71,211 @@ export function PlayerList() {
...
+              <Popover.Content>
+                <Flex direction="column" gap="2">
+                  <PlayerMenuItem
+                    id="rename"
+                    label="Rename"
+                    icon={MdDriveFileRenameOutline}
+                    onClick={() => {
+                      /* TODO */
+                    }}
+                    disabled={true}
```
Comment (vermentor) [inline client/src/GamemasterInGame/PlayerList.tsx:89]: Just an idea I had and wanted to remember

File: client/src/GamemasterInGame/Lobby.tsx [hunk 4]
```diff
@@ -101,10 +105,10 @@ export function Lobby({ rolesList }: LobbyProps) {
                     gap="2"
                     style={{ overflowY: "auto" }}
                   >
-                    <GameMasterActions
-                      selectedRoles={characterSelectState.selectedRoles.value}
```
Comment (vermentor) [inline client/src/GamemasterInGame/Lobby.tsx:105]: Was showing twice, so removed the duplicate buttons.

File: client/src/GamemasterInGame/NightOrder.tsx
Comment (vermentor) [inline client/src/GamemasterInGame/NightOrder.tsx:1]: Still cleaning this component up

File: client/src/GamemasterInGame/Lobby.tsx (hunk 1 above)
Comment (Alphy11) [reply client/src/GamemasterInGame/Lobby.tsx:77]: .filter(Boolean)

File: client/src/GamemasterInGame/Lobby.tsx (hunk 4 above)
Comment (Alphy11) [reply client/src/GamemasterInGame/Lobby.tsx:105]: ah merge conflict

File: client/src/GamemasterInGame/NightOrder.tsx (hunk 2 above)
Comment (Alphy11) [reply client/src/GamemasterInGame/NightOrder.tsx:13]: dunno what you were foing wrong.  But I like not passing props and letting components control things

**Pull request merged**